packages/examples/src/chat-function-call.tsx

```diff
@@ -0,0 +1,65 @@
+import * as LLMx from 'ai-jsx';
+import { showInspector } from 'ai-jsx/core/inspector';
+import {
+  ChatCompletion,
+  ChatProvider,
+  SystemMessage,
+  UserMessage,
+  FunctionCall,
+  FunctionResponse,
+} from 'ai-jsx/core/completion';
+
+function ModelProducesFunctionCall({ query }: { query: string }) {
+  return (
+    <ChatProvider model="gpt-4-0613">
+      <ChatCompletion
+        functionDefinitions={[
+          {
+            name: 'evaluate_expression',
+            description: 'Evaluates a mathematical expression',
+            parameters: {
+              expression: {
+                description: 'The mathematical expression to be evaluated.',
+                type: 'string',
+                required: true,
+              },
+            },
+          },
+        ]}
+      >
+        <SystemMessage>You are a tool that may use functions to answer a user question.</SystemMessage>
+        <UserMessage>{query}</UserMessage>
+      </ChatCompletion>
+    </ChatProvider>
+  );
+}
+
+function ModelProducesFinalResponse({ query }: { query: string }) {
+  return (
+    <ChatProvider model="gpt-4-0613">
+      <ChatCompletion
+        functionDefinitions={[
+          {
+            name: 'evaluate_expression',
+            description: 'Evaluates a mathematical expression',
+            parameters: {
+              expression: {
+                description: 'The mathematical expression to be evaluated.',
+                type: 'string',
+                required: true,
+              },
+            },
+          },
+        ]}
+      >
+        <SystemMessage>You are a tool that may use functions to answer a user question.</SystemMessage>
+        <UserMessage>{query}</UserMessage>
+        <FunctionCall name="evaluate_expression" args={{ expression: '2523231 * 2382382' }} />
+        <FunctionResponse name="evaluate_expression">6011300116242</FunctionResponse>
+      </ChatCompletion>
+    </ChatProvider>
+  );
+}
+
+showInspector(<ModelProducesFunctionCall query="What is 2523231 * 2382382?" />);
```

Comment:
@petersalas from this example, we see the output is:

```tsx
<FunctionCall name="evaluate_expression" args={{"expression":"2523231 * 2382382"}} />
```

I was thinking the best way to use this from `UseTools` would be to `partialRender` through to a `FunctionCall`s, then inspect the `props` to figure out which funcs to invoke. Does that match what you were thinking? If so, I'm comfortable merging this PR as we have a clear path forward.

Reply:
I guess this is what `ChatCompletion` is doing already, and it seems good, so 👍